src/Symfony/Component/DependencyInjection/Loader/Configurator/ContainerConfigurator.php

```diff
@@ -125,8 +125,12 @@ function service_locator(array $values): ServiceLocatorArgument
 {
     $values = AbstractConfigurator::processValue($values, true);
 
-    if (isset($values[0])) {
-        trigger_deprecation('symfony/dependency-injection', '6.3', 'Using integers as keys in a "service_locator()" argument is deprecated. The keys will default to the IDs of the original services in 7.0.');
+    foreach ($values as $k => $v) {
+        if (\is_int($k)) {
```

Comment:
can't this concern be dealt with in ServiceLocatorTagPass?

Reply:
I'd say yes, but given that depreciation was placed outside `ServiceLocatorTagPass`, doesn't this bring about a change in behavior in a place where it wasn't anticipated?

It also means that `PhpFileLoaderTest::testServiceWithServiceLocatorArgument()` continues to accept and return numeric keys because the compiler pass is not used there. I feel that this comes down to removing the expected deprecation and keeping the previous behavior. :thinking:

Reply:
I pushed what I mean here.

Reply:
Thank you!